src/main/java/io/github/mountainrange/mule/gameplay/WorldMap.java

```diff
@@ -10,21 +11,28 @@
  * A class to represent the map and facilitates interactions
  * between the actual data store and the rest of the program
  */
-public class WorldMap implements Iterable<Tile> {
+public class WorldMap implements Iterable<VisualTile> {
```

Comment:
Maybe we could rename `TileInterface` to just `Tile`, and make this `Iterable<Tile>`, so that it hides the visual stuff from `GameManager`.